airgun/entities/product.py

```diff
@@ -124,3 +173,35 @@ def prerequisite(self, *args, **kwargs):
 
     def step(self, *args, **kwargs):
         self.parent.repo_discovery.click()
+
+
+@navigator.register(ProductEntity, 'Select Action')
+class ProductsSelectAction(NavigateStep):
+    """Navigate to Action page by selecting checkboxes for necessary Products and
+     then clicking on the action name button in 'Select Action' dropdown.
+
+    Args:
+        action_name: the action name to select from dropdown button
+        entities_list: list of Products that need to be modified
+    """
+    ACTIONS_VIEWS = {
+        'Manage HTTP Proxy': ProductManageHttpProxy,
+        'Sync Selected': ProductSyncSelected,
+    }
+
+    def prerequisite(self, *args, **kwargs):
+        return self.navigate_to(self.obj, 'All')
+
+    def step(self, *args, **kwargs):
+        action_name = kwargs.get('action_name')
+        self.VIEW = self.ACTIONS_VIEWS.get(action_name)
+        if not self.VIEW:
+            raise ValueError('Please provide a valid action name.'
+                             ' action_name: "{0}" not found.')
+        entities_list = kwargs.get('entities_list')
+        if entities_list == "All":
+            self.parent.select_all.click()
+        else:
+            for entity in entities_list:
+                self.parent.table.row(name=entity)[0].click()
```

Comment:
You can't do that, because `ProductsTableView` class doesn't have `select_all` element. If we call `session.product.manage_http_proxy('All', ...)`, we will get exception here.

You can take a look at https://github.com/SatelliteQE/airgun/pull/437 for all the details, or just drop support for `'All'` and only leave `for` loop here.

Reply:
Dropped support for `'All'`